### ISSUE TITLE
Android: Kotest-assertions uses java.nio.file.Path causing failure on Android < 26

### ISSUE THREAD

**Issue by JustinTullgren** [kotest/kotest#1563]
**Kotest Version**: 4.1.1
**Kotest Library**: kotest-assertions-core-jvm
**Environment**: Android Api <26 (Android 8 (O)reo)

Hi,

I was very happy to see the assertions library was broken out into its own library and attempted to use it in an "Android" test (which requires the test to be run on an android phone) so we could be consistent with our unit tests at least for assertions.

The assertions work fine for Android O and above (android api 26).

However, the `platformShow` actual implementation in the _jvmMain_ source uses `java.nio.file.Path` which is unfortunately not available on older android devices. Printing messages for failures results in runtime errors of class not found.

Would it be possible to use the `java.io.File#getPath` to print instead of calling toString on the `java.nio.file.path`?

Here is the [page](https://developer.android.com/reference/java/nio/file/Path) which shows that `Path` was added in api 26, meaning it cannot be used on older devices for testing.

Here is the [page](https://developer.android.com/reference/java/io/File#getPath()) which shows you can get the file path on older android devices back to Api 9 (very old).

```
@file:JvmName("platformjvm")
package io.kotest.assertions.show

import java.nio.file.Path

actual fun <A : Any> platformShow(a: A): Show<A>? = when (a) {
   is Path -> PathShow as Show<A>
   else -> null
}
```
 ([actual platform show](https://github.com/kotest/kotest/blob/master/kotest-assertions/kotest-assertions-shared/src/jvmMain/kotlin/io/kotest/assertions/show/platformShow.kt#L4))

```
package io.kotest.assertions.show

import java.nio.file.Path

object PathShow : Show<Path> {
   override fun show(a: Path): Printed = a.toString().printed()
}
```
([pathshow](https://github.com/kotest/kotest/blob/master/kotest-assertions/kotest-assertions-shared/src/jvmMain/kotlin/io/kotest/assertions/show/PathShow.kt#L3))

`expect fun <A : Any> platformShow(a: A): Show<A>?` ([expect platform show](https://github.com/kotest/kotest/blob/master/kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/show/Show.kt#L32))

**Comment by sksamuel** [comment]
Create the PR and I'll merge it and include it in 4.1.2

**Comment by LeoColman** [comment]
Sure! 
It seems really easy to make a change. 
Do you want to try to do it yourself?

**Comment by JustinTullgren** [comment]
@LeoColman I can try. I am not sure how to properly contribute though. I only see release/4.0.x and nothing about 4.1.x according to github.  If I understand the contributing page correctly i should branch off a "4.1.x" branch correct?  Also do I need to fork and perform an upstream pr?

Any guidance would be appreciated.

Also I didn't see anything about running tests.  Can i just run gradlew test on the jvm module or no? 

Thanks.

**Comment by sksamuel** [comment]
You can run ./gradlew check
You can fork the repo and do the PR based on master (which is 4.1 dev)

**Comment by JustinTullgren** [comment]
@sksamuel running check for the assertions library on master stalls at 77 tests passing, it just hangs. Is this known to you? 

Also, I published the jvm publications to maven local with my changes but the maven metadata doesn't seem to be resolving transitive dependencies correctly for me when using mavenLocal in a consuming android project. `Failed to resolve: io.kotest:kotest-assertions-shared:4.2.0.0-SNAPSHOT` (this artifact and POM exists in my local .m2)

I see the artifacts in my ~/.m2 and have added mavenLocal() first in all my repositories in my consuming app yet no luck.  

I have a fix but I just don't feel confident since the `check` hangs and I can't actually publish to my maven local and test in a consuming app.  Is there way for you to publish this fix and for me to test in a snapshot release?

**Comment by sksamuel** [comment]
I can run `./gradlew check` and all passes on master at root.
It also passes on the server:
https://github.com/kotest/kotest/actions

Master is currently kotlin 1.4 m3. Maybe you'll have better look against
the release/4.1.x branch which is kotlin 1.3.72
Perhaps your system doesn't like the kotlin version.

**Comment by JustinTullgren** [comment]
@sksamuel That was the issue. I thought I opened the project up in Intellij EAP but had not.  The check task runs completely now and successfully. Here is the PR https://github.com/kotest/kotest/pull/1572

cc @LeoColman

**Comment by sksamuel** [comment]
Will be included in 4.1.3
